### ISSUE TITLE
TBB/2018_U5@conan/stable fails to build on Windows (AppVeyor) using clang-6

### ISSUE THREAD

**Issue by acgetchell** [conan-community/community#43]
### Description of Problem, Request, or Question

TBB/2018_U5@conan/stable fails to build on Windows (AppVeyor) using clang-6:

https://ci.appveyor.com/project/acgetchell/cdt-plusplus/build/job/6x679c9m5taw8p2m#L3442

### Package Details (Include if Applicable)
* Package Name/Version: **TBB/2018_U5**
* Operating System: **Windows**
* Compiler+version: **clang-6**

### Steps to reproduce (Include if Applicable)

My AppVeyor file is here:

https://github.com/acgetchell/CDT-plusplus/blob/develop/.appveyor.yml

Conan:

https://github.com/acgetchell/CDT-plusplus/blob/develop/conanfile.py

CMake:

https://github.com/acgetchell/CDT-plusplus/blob/develop/CMakeLists.txt

It works on Linux/MacOS with Travis-CI:

https://travis-ci.org/acgetchell/CDT-plusplus

**Comment by acgetchell** [comment]
I also tried building it with `arch=x86`

https://ci.appveyor.com/project/acgetchell/cdt-plusplus/build/0.1.8.567/job/85un1quvb40p69ae#L3470

**Comment by acgetchell** [comment]
It actually doesn't work on MacOS either:

https://github.com/conan-community/community/issues/45

**Comment by SSE4** [comment]
this one was tricky.
TBB detects RTM (Restricted Transactional Memory) intrinsic compiler-support (functions like `_xabort`, `_xbegin`, `_end`) using the following code:
```
#define __TBB_TSX_INTRINSICS_PRESENT ((__RTM__ || _MSC_VER>=1700 || __INTEL_COMPILER>=1300) && !__TBB_DEFINE_MIC && !__ANDROID__)
```
it means, either if `__RTM__` macro is defined (it is defined for Clang and GCC when you pass `-mrtm` to the command line, similarly to `-msse` and friends), or if `_MSC_VER >= 1700` (literally, Visual Studio 2012 or higher)
when **clang-cl** comes into play, it tries to emulate MSVC as much as possible and defines `_MSC_VER` as well (version is controlled by `-fms-compatibility-version` flag).
now, let's take a look at `C:\Program Files\LLVM\lib\clang\6.0.1\include\immintrin.h`:
```
#if !defined(_MSC_VER) || __has_feature(modules) || defined(__RTM__)

#include <rtmintrin.h>
```
so, RTM intrinsics are available only if _MSC_VER is not defined at all (e.g. if you pass `-fno-ms-compatibility`), or __RTM__ is define (`-mrtm` passed).
therefore, it's either TBB's detection bug, or clang-cl bug, not sure which one.
TLDR:
you may use the following to work-around the issue:
```
CC=clang-cl.exe -mrtm
CXX=clang-cl.exe -mrtm
```

**Comment by acgetchell** [comment]
Thanks, that fixed that problem. 

New problem with /SAFESEH on x64 machines, which is odd because I specified x86.

https://github.com/acgetchell/CDT-plusplus/blob/fe53b797864c940aed15b8b2c3546ac1da0479b2/.appveyor.yml#L90

https://ci.appveyor.com/project/acgetchell/cdt-plusplus/build/0.1.8.568/job/9ukuaw1jvubq437x

**Comment by SSE4** [comment]
interesting, I'll take a look

**Comment by acgetchell** [comment]
Okay, it builds completely with x64:

https://ci.appveyor.com/project/acgetchell/cdt-plusplus/build/job/pdp39bf0nx41ao6m?fullLog=true#L5605

AppVeyor is erroring out because of date/2.4@bincrafters/stable in VS2015 and bzip2/1.0.6@conan/stable in VS2017, but TBB is building in both environments.
